Sources/PackageGraph/RawPackageConstraints.swift

```diff
@@ -13,12 +13,13 @@ import SourceControl
 
 extension PackageDependencyDescription {
     /// Create the package reference object for the dependency.
-    public func createPackageRef(config: SwiftPMConfig) -> PackageReference {
+    public func createPackageRef(config: SwiftPMConfig, isRoot: Bool = false) -> PackageReference {
         let effectiveURL = config.mirroredURL(forURL: self.url)
         return PackageReference(
             identity: PackageReference.computeIdentity(packageURL: effectiveURL),
             path: effectiveURL,
-            isLocal: (requirement == .localPackage)
+            isLocal: (requirement == .localPackage),
+            isRoot: isRoot
```

Comment:
maybe we should make a reference kind enum to represent local, remote and root references?

Reply:
Agreed. That seems like a cleaner solution.